base/system/winlogon/CMakeLists.txt

```diff
@@ -1,3 +1,6 @@
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    add_compile_options(-Wno-missing-braces)
+endif()
```

Comment:
Please avoid this. We add such hacks only to 3rd party modules.

Reply:
It is a known GCC bug from what I can tell, you cannot assign an `SID_IDENTIFIER_AUTHORITY` to a well-known SID without throwing this error. In this case it says that `SECURITY_NT_AUTHORITY` is missing braces, which is incorrect.

Reply:
@cbialorucki FYI to understand why you got the "missing braces" error (looks like your comment suddenly vanished, did  you delete it?). `SID_IDENTIFIER_AUTHORITY` is a structure with identifier field being a fixed array of 6 elements and authority identifiers are initialized in the following way:

https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/xdk/setypes.h;hb=f1549bc251c626cc90751d7b7b59a4370c0e4c98#l522

Therefore initializing `SID_IDENTIFIER_AUTHORITY NtAuthority = SECURITY_NT_AUTHORITY;` as you do now in code is like doing `NtAuthority = {0,0,0,0,0,5};` once the compiler scraps human-readable constructs. The proper fix is to just wrap the authority constant like we do in 99% of security related code in ROS.

https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=%7BSECURITY_NT_AUTHORITY%7D

While we tend to mock GCC quite a lot (and rightfully so at times), this error is not unreasonable.